test/body.test.ts

```diff
@@ -48,14 +84,34 @@ describe('', () => {
       expect(result.text).toBe('200')
     })
 
-    it('handles empty body', async () => {
+    it('handles non-present body', async () => {
       let _body = 'initial'
       app.use('/', eventHandler(async (request) => { _body = await readBody(request); return '200' }))
       const result = await request.post('/api/test').send()
       expect(_body).toBeUndefined()
       expect(result.text).toBe('200')
     })
 
+    it('handles empty body', async () => {
+      let _body = 'initial'
+      app.use('/', eventHandler(async (request) => {
+        _body = await readBody(request); return '200'
+      }))
+      const result = await request.post('/api/test').set('Content-Type', 'text/plain').send('')
+      expect(_body).toStrictEqual({})
```

Comment:
Now i see what you mean 👍🏼  This seems a wrong behavior from`h3.readBody` that empty string results in empty object.